indexers/accounts/accounts.js

```diff
@@ -0,0 +1,153 @@
+
+    if (!profileData) {
+      return;
+    }
+
+    try {
+      console.log(
+        `Attemp to store account data for account and fields:  ${Object.keys(
+          profileAction
+        ).join(", ")}`
+      );
+      const fields = Object.keys(profileData);
+      const valuesToStrings = {};
+
+      fields.forEach((field) => {
+        switch (field) {
+          case "image":
+          case "linktree":
+          case "tags":
+            valuesToStrings[field] = JSON.stringify(profileData[field]);
+            break;
+          case "backgroundImage":
+            valuesToStrings.background_image = JSON.stringify(
+              profileData.backgroundImage
+            );
+            break;
+          case "horizon_tnc":
+            valuesToStrings.horizon_tnc = JSON.parse(profileData.horizon_tnc);
+            break;
+          default:
+            valuesToStrings[field] = profileData[field];
+            break;
+        }
+      });
+
+      await context.db.Accounts.upsert(
+        {
+          account_id: accountId,
+          block_height: blockHeight,
+          block_timestamp_ms: blockTimestamp,
+          ...valuesToStrings,
+        },
+        ["account_id"],
+        Object.keys(valuesToStrings)
+      );
+    } catch (error) {
+      console.log(`Failed to store data for ${accountId}. Error ${error}`);
+    }
+  }
+
+  let accountProfileActions = [];
+  try {
+    const actions = block.actions();
+
+    if (!actions) {
+      console.log("Block has no actions");
+      return;
+    }
+    const contractActions = actions.filter(
+      (action) => action.receiverId === SOCIAL_DB
+    );
+    if (!contractActions) {
+      console.log("Block has no actions");
+      return;
+    }
+    accountProfileActions = contractActions.flatMap((action) =>
+      action.operations
+        .map((operation) => operation["FunctionCall"])
+        .filter((operation) => operation?.methodName === "set")
+        .map((functionCallOperation) => {
+          try {
+            return {
+              ...functionCallOperation,
+              args: base64decode(functionCallOperation.args),
+              receiptId: action.receiptId, // providing receiptId as we need it
+            };
+          } catch (e) {
+            console.log("Error parsing function call", e);
+          }
+        })
+        .filter((functionCall) => {
+          try {
+            if (
+              !functionCall ||
+              !functionCall.args ||
+              !functionCall.args.data ||
+              !Object.keys(functionCall.args.data) ||
+              !Object.keys(functionCall.args.data)[0]
+            ) {
+              console.log(
+                "Set operation did not have arg data in expected format"
+              );
+              return;
+            }
+            const accountId = Object.keys(functionCall.args.data)[0];
+            return Object.keys(functionCall.args.data[accountId]).includes(
```

Comment:
We probably need another check to see whether `functionCall.args.data[accountId]` exists.